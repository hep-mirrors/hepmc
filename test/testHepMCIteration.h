//////////////////////////////////////////////////////////////////////////
// testHepMCIteration.h
//
// garren@fnal.gov, May 2007
//
// Define methods and classes used by testHepMCIteration
//////////////////////////////////////////////////////////////////////////

/// returns true if the GenParticle particle is a photon with pT > 10 GeV
bool IsPhoton( const HepMC::GenParticle* p ) { 
    if ( p->pdg_id() == 22 
	 && p->momentum().perp() > 10. ) return true;
    return false;
}

/// returns true if the GenParticle is a W+/W-
bool IsWBoson( const HepMC::GenParticle* p ) { 
    if ( abs(p->pdg_id()) == 24 ) return true;
    return false;
}

//! test class

/// \class  IsFinalState
/// this predicate returns true if the input has no decay vertex
class IsFinalState {
public:
    /// returns true if the GenParticle does not decay
    bool operator()( const HepMC::GenParticle* p ) { 
	if ( !p->end_vertex() && p->status()==1 ) return true;
	return false;
    }
};


