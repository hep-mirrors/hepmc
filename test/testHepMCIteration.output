Processing Event Number 1 its # 1
3 out of 34 processed events passed the cuts. Finished.
