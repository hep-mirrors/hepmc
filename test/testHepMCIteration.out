Processing Event Number 1 its # 1
Event Number 34 is good 
Processing Event Number 51 its # 51
Event Number 79 is good 
Processing Event Number 100 its # 100
2 out of 100 processed events passed the cuts. Finished.
